asciidoc-confluence-publisher-converter/src/main/java/org/sahli/asciidoc/confluence/publisher/converter/FolderBasedAsciidocPagesStructureProvider.java

```diff
@@ -0,0 +1,124 @@
+    private static final String INCLUDE_FILE_PREFIX = "_";
+
+    private final AsciidocPagesStructure structure;
+
+    public FolderBasedAsciidocPagesStructureProvider(Path documentationRootFolder) {
+        this.structure = buildStructure(documentationRootFolder);
+    }
+
+    @Override
+    public AsciidocPagesStructure structure() {
+        return this.structure;
+    }
+
+    private AsciidocPagesStructure buildStructure(Path documentationRootFolder) {
+        try {
+            Map<Path, DefaultAsciidocPage> asciidocPageIndex = indexAsciidocPagesByFolderPath(documentationRootFolder);
+            List<DefaultAsciidocPage> allAsciidocPages = connectAsciidocPagesToParent(asciidocPageIndex);
+            List<AsciidocPage> topLevelAsciiPages = findTopLevelAsciiPages(allAsciidocPages, documentationRootFolder);
+
+            return new DefaultAsciidocPagesStructure(topLevelAsciiPages);
+        } catch (IOException e) {
+            throw new RuntimeException("Could not create asciidoc source structure", e);
+        }
+    }
+
+    @SuppressWarnings("CodeBlock2Expr")
+    private List<DefaultAsciidocPage> connectAsciidocPagesToParent(Map<Path, DefaultAsciidocPage> asciidocPageIndex) {
+        asciidocPageIndex.forEach((asciidocPageFolderPath, asciidocPage) -> {
+            asciidocPageIndex.computeIfPresent(asciidocPage.path().getParent(), (ignored, parentAsciidocPage) -> {
+                parentAsciidocPage.addChild(asciidocPage);
+
+                return parentAsciidocPage;
+            });
+        });
+
+        return new ArrayList<>(asciidocPageIndex.values());
+    }
+
+    private static Map<Path, DefaultAsciidocPage> indexAsciidocPagesByFolderPath(Path documentationRootFolder) throws IOException {
+        return walk(documentationRootFolder)
+                .filter((path) -> isAdocFile(path) && !(isIncludeFile(path)))
+                .collect(toMap((asciidocPagePath) -> removeExtension(asciidocPagePath), (asciidocPagePath) -> new DefaultAsciidocPage(asciidocPagePath)));
+    }
+
+    private static List<AsciidocPage> findTopLevelAsciiPages(List<DefaultAsciidocPage> asciiPageByFolderPath, Path documentationRootFolder) {
+        return asciiPageByFolderPath.stream()
+                .filter((asciidocPage) -> asciidocPage.path().equals(documentationRootFolder.resolve(asciidocPage.path().getFileName())))
+                .collect(toList());
+    }
+
+    private static Path removeExtension(Path path) {
+        return Paths.get(path.toString().substring(0, path.toString().lastIndexOf('.')));
+    }
+
+    private static boolean isAdocFile(Path file) {
+        return file.toString().endsWith(ADOC_FILE_EXTENSION);
+    }
+
+    private static boolean isIncludeFile(Path file) {
+        return file.getFileName().toString().startsWith(INCLUDE_FILE_PREFIX);
+    }
+
+
+    private static class DefaultAsciidocPage implements AsciidocPage {
+
+        private final Path path;
+        private final List<AsciidocPage> children;
+
+
```

Comment:
One line break to much ;-)

Reply:
Fixed in 4dc419e18e3a2cfc8f72f87a6f2a370c7e58f27a